packages/ra-ui-materialui/src/button/PrevNextButton.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import {
+    ListParams,
+    SORT_ASC,
+    useCreatePath,
+    useGetList,
+    useRecordContext,
+    useResourceContext,
+    useStore,
+    useTranslate,
+} from 'ra-core';
+import { NavigateBefore, NavigateNext } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { IconButton, SxProps, styled } from '@mui/material';
+
+export const PrevNextButton = (props: PrevNextButtonProps) => {
+    const {
+        linkType = 'edit',
+        sx,
+        storeKey,
+        limit = 1000,
+        listParams = defaultParams,
+        staleTime = 5 * 60 * 1000,
+    } = props;
+
+    const translate = useTranslate();
+    const record = useRecordContext();
+    const resource = useResourceContext();
+    const createPath = useCreatePath();
+    const [storedParams] = useStore<Params>(
+        storeKey || `${resource}.listParams`,
+        defaultParams
+    );
+
+    const { isLoading, data, isError, error } = useGetList(
+        resource,
+        {
+            sort: {
+                ...{ field: storedParams.sort, order: storedParams.order },
+                ...{ field: listParams.sort, order: listParams.order },
+            },
+            filter: { ...storedParams.filter, ...listParams.filter },
+            ...(limit ? { pagination: { page: 1, perPage: limit } } : {}),
```

Comment:
I think the pagination is compulsory. I wouldn't support the case where there is no limit. If people want an unlimited request, they should set an arbitrary large limit.